flocker/node/agents/test/test_blockdevice.py

```diff
@@ -0,0 +1,459 @@
+        )
+        self.assertEqual([expected_volume], self.api.list_volumes())
+
+    def test_list_attached_and_unattached(self):
+        """
+        ``list_volumes`` returns both attached and unattached
+        ``BlockDeviceVolume``s.
+        """
+        expected_host = b'192.0.2.123'
+        new_volume1 = self.api.create_volume(size=1024)
+        new_volume2 = self.api.create_volume(size=1024)
+        attached_volume = self.api.attach_volume(
+            blockdevice_id=new_volume2.blockdevice_id,
+            host=expected_host
+        )
+        self.assertEqual(
+            sorted_by_blockdevice_id([new_volume1, attached_volume]),
+            sorted_by_blockdevice_id(self.api.list_volumes())
+        )
+
+    def test_multiple_volumes_attached_to_host(self):
+        """
+        ``attach_volume`` can attach multiple block devices to a single host.
+        """
+        expected_host = b'192.0.2.123'
+        volume1 = self.api.create_volume(size=1024)
+        volume2 = self.api.create_volume(size=1024)
+        attached_volume1 = self.api.attach_volume(
+            volume1.blockdevice_id, host=expected_host
+        )
+        attached_volume2 = self.api.attach_volume(
+            volume2.blockdevice_id, host=expected_host
+        )
+
+        self.assertEqual(
+            sorted_by_blockdevice_id([attached_volume1, attached_volume2]),
+            sorted_by_blockdevice_id(self.api.list_volumes())
+        )
+
+    def test_get_device_path_unknown_volume(self):
+        """
+        ``get_device_path`` raises ``UnknownVolume`` if the supplied
+        ``blockdevice_id`` has not been created.
+        """
+        unknown_blockdevice_id = unicode(uuid4())
+        exception = self.assertRaises(
+            UnknownVolume,
+            self.api.get_device_path,
+            unknown_blockdevice_id
+        )
+        self.assertEqual(unknown_blockdevice_id, exception.blockdevice_id)
+
+    def test_get_device_path_unattached_volume(self):
+        """
+        ``get_device_path`` raises ``UnattachedVolume`` if the supplied
+        ``blockdevice_id`` corresponds to an unattached volume.
+        """
+        new_volume = self.api.create_volume(size=1000)
+        exception = self.assertRaises(
+            UnattachedVolume,
+            self.api.get_device_path,
+            new_volume.blockdevice_id
+        )
+        self.assertEqual(new_volume.blockdevice_id, exception.blockdevice_id)
+
+    def test_get_device_path_device(self):
+        """
+        ``get_device_path`` returns a ``FilePath`` to the device representing
+        the attached volume.
+        """
+        new_volume = self.api.create_volume(size=1024 * 50)
+        attached_volume = self.api.attach_volume(
+            new_volume.blockdevice_id,
+            b'192.0.2.123'
+        )
+        device_path = self.api.get_device_path(attached_volume.blockdevice_id)
+        self.assertTrue(
+            device_path.isBlockDevice(),
```

Comment:
The idea is you're attaching block devices to some other machine. The path is therefore the path on that other machine, not locally... so `isBlockDevice()` isn't useful for real block devices.

Maybe... create a filesystem on the device? Of course if you have bugs that may destroy something :worried: